src/modules/admin/layout/AdminSideNav.tsx

```diff
@@ -43,22 +36,41 @@ const AdminSideNav: React.FC<AdminSideNavProps> = ({ currentPageId }) => {
     [adminAddEditPromotion, adminEditPoint, adminEditCoupon],
     adminSalesReport,
     [adminSetting, adminUserManagement],
-    employeeStock,
     employeeOrderManagement,
+    employeeStock,
     accountManagement,
   ];
 
-  const items: MenuProps["items"] | MenuProps["items"][] = adminPages.map(
-    (page) => {
+  const { data: user } = useUser();
+
+  const getItem = (
+    page: PageMetaData,
+    children?: PageMetaData[],
+  ): MenuItem | null => {
+    if (!page) return null;
+
+    if (roleToRoleNumber[page.minimumRole ?? "Customer"] <= (user?.role ?? 1)) {
+      return {
+        key: page.id,
+        icon: page.Icon && <page.Icon />,
+        label: page.label,
+        children: children?.map((page) => getItem(page)),
+      };
+    }
+    return null;
+  };
+
+  const items: MenuProps["items"] | MenuProps["items"][] = adminPages
+    .map((page) => {
       if (Array.isArray(page)) {
         return getItem(
           page[0] as PageMetaData,
           page.slice(1) as PageMetaData[],
         );
       }
       return getItem(page);
-    },
-  );
+    })
+    .filter((item) => Boolean(item));
```

Comment:
Why we need this line.

Reply:
const getItem = (
    page: PageMetaData,
    children?: PageMetaData[],
  ): MenuItem | null 
  
getItem may return null so we need this to filter null out

Reply:
> Why we need this line, please give me some example

+1